src/controller/CHIPDevice.h

```diff
@@ -349,6 +351,13 @@ class DLL_EXPORT Device : public Messaging::ExchangeDelegate, public SessionEsta
 
     CHIP_ERROR GenerateCSRNonce() { return Crypto::DRBG_get_bytes(mCSRNonce, sizeof(mCSRNonce)); }
 
+    CHIP_ERROR SetCSRNonce(ByteSpan csrNonce)
+    {
+       memcpy(mCSRNonce, csrNonce.data(), csrNonce.size());
```

Comment:
Need to check that csrNonce.size == sizeof(mCSRNonce).

If they are not equal, return error. It looks like the GetCSRNonce relies that this is the exact size and we do not want to access uninitialized memory otherwise.

Reply:
Done